### PR TITLE
571 handle some Sage failures with retries

### PR DESCRIPTION
**Pull request by hwindsor**
## Pull Request Overview

- Implements a retry mechanism for Sage requests with limits on numbers of retries
- Provides a default id_config from teh ia_json if none provided by the user

---

### CODE REVIEW

File: migrations/versions/fd467d5e0812_.py [hunk 1]
```diff
@@ -0,0 +1,50 @@
...
+
+
+def upgrade():
+    """
+    Upgrade Semantic Description:
+        ENTER DESCRIPTION HERE
+    """
+    # ### commands auto generated by Alembic - please adjust! ###
+    with op.batch_alter_table('asset_group_sighting', schema=None) as batch_op:
+        batch_op.add_column(sa.Column('detection_attempts', sa.Integer(), nullable=False))
```
Comment (naknomum) [inline migrations/versions/fd467d5e0812_.py:28]: having a default value on the column `0` is the only suggestion i have here.  save broken migrations on existing data in `asset_group_sighting` instances.

**Review by naknomum — APPROVED** [review]
looks good to me; just the one comment about default=0 on db table migration.

File: migrations/versions/fd467d5e0812_.py (hunk 1 above)
Comment (hwindsor) [reply migrations/versions/fd467d5e0812_.py]: Agreed, done, with a quoted value which feels utterly wrong

**Review by naknomum — APPROVED** [review]
ok, `server_default` seems good in migration now.  :+1:

**Pull request merged**